### DISCUSSION TITLE
How can I display the current selected story when remixing the docs page?

### DISCUSSION THREAD

**Discussion by drik98** [storybookjs/storybook#17216] [Help] [answered]
Currently I use 

```javascript
import {
   Title,
   Subtitle,
   Description,
   Primary,
   ArgsTable,
   PRIMARY_STORY,
 } from '@storybook/addon-docs';

export const parameters = {
   docs: {
      page: () => (
         <>
           <Title />
           <Subtitle />
           <Description />
           <Primary />
           <ArgsTable story={PRIMARY_STORY} />
         </>
      )
   }
};
```

in order to remix the docs page.

What I want to achieve is that not all stories are displayed on the docs page but only the currently selected one.

At the moment the Primary story is display no matter what. What I imagine should be possible is something like:

```javascript
export const parameters = {
   docs: {
      page: () => (
         <>
           <Title />
           <Subtitle />
           <Description />
           <Story id={CURRENT_STORY} />
           <ArgsTable story={CURRENT_STORY} />
         </>
      )
   }
};
```

Any ideas?

**Comment by shilman** [comment]
It's `CURRENT_SELECTION` not `CURRENT_STORY`. Otherwise that should work.

**Reply by drik98** [reply]
Thanks for your reply. This almost seems like what I am trying to achieve.

However: Somehow it seems to be incomplete. There is no border around the Story and the "Show code" button is missing. Am I missing something?
<img width="1053" alt="primary" src="https://user-images.githubusercontent.com/36853386/150743475-4e67c36f-e156-4f9d-be1c-d65d448465ae.png">
<img width="1376" alt="current_selection" src="https://user-images.githubusercontent.com/36853386/150743481-729ad8ef-7493-4015-8c1b-bdc277d538a6.png">

**Reply by drik98** [reply]
@shilman any idea?

**Comment by siminino** [comment]
Hi @drik98, it seems the Story component does not have the wrapping features included by Primary (and so DocsStory) component.

I could achieve what you are looking for by using Primary with `context.name`, it seems to store the current story name:
```jsx
const DocsPage = () => {
  const context = useContext(DocsContext)
  return (
    <>
      <Title />
      <Subtitle />
      <Description />
      <Primary name={context.name} />
      <ArgsTable story={CURRENT_SELECTION} />
    </>
  )
}
```

But this is not all. Now I am having another issue. When I navigate to another story of the same component, docs args is not refresh in PropsTable, displaying the args of the previous story.

**Comment by siminino** [comment, marked as answer]
Hi @drik98, i found a little trick to solve the issue i mentioned before. Passing the current story name has key to ArgsTable is going to make React render a new ArgsTable, ignoring the state of the rendered before.

```jsx
const DocsPage = () => {
  const context = useContext(DocsContext)
  return (
    <>
      <Title />
      <Subtitle />
      <Description />
      <Primary name={context.name} />
      <ArgsTable key={context.name} story={CURRENT_SELECTION} />
    </>
  )
}
```
With this change, i have the behaviour i expect.

**Reply by drik98** [reply]
Thanks a lot! Thats exactly what I was trying to accomplish.

**Comment by drik98** [comment]
For the sake of completeness for any one else dropping by here having the same problem. Heres my full code snippet including imports for clarity.
```javascript
import {
   ArgsTable,
   CURRENT_SELECTION,
   Description,
   DocsContext,
   Primary,
   Subtitle,
   Title
} from '@storybook/addon-docs';
import { useContext } from 'react';

const DocsPage = () => {
  const context = useContext(DocsContext);
  return (
    <>
        <Title />
        <Subtitle />
        <Description />
        <Primary name={context.name} />
        <ArgsTable key={context.name} story={CURRENT_SELECTION} />
    </>
  )
}
```
